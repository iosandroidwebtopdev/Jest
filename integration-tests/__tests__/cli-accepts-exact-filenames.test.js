/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

'use strict';

const path = require('path');
const skipOnWindows = require('../../scripts/skip_on_windows');
const {extractSummary, cleanup, writeFiles} = require('../utils');
const runJest = require('../runJest');

const DIR = path.resolve(__dirname, '../cli_accepts_exact_filenames');

skipOnWindows.suite();

beforeEach(() => cleanup(DIR));
afterAll(() => cleanup(DIR));

test('CLI accepts exact filenames', () => {
  writeFiles(DIR, {
    'bar.js': `
      test('bar', () => console.log('Hey'));
    `,
    'foo/baz.js': `
      test('baz', () => console.log('Hey'));
    `,
    'package.json': '{}',
  });

  const {stderr, stdout, status} = runJest(DIR, [
    '-i',
    '--ci=false',
    '--forceExit',
    './bar.js',
    './foo/baz.js',
    './foo',
  ]);

  expect(status).toBe(0);

  const {rest, summary} = extractSummary(stderr);

  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
  expect(stdout).toMatchSnapshot();
});
