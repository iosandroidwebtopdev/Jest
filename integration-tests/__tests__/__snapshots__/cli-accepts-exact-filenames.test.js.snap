// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI accepts exact filenames 1`] = `
"PASS ./bar.js
  ● Console

    console.log bar.js:2
      Hey

PASS foo/baz.js
  ● Console

    console.log foo/baz.js:2
      Hey


"
`;

exports[`CLI accepts exact filenames 2`] = `
"Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites matching /.\\\\/bar.js|.\\\\/foo\\\\/baz.js|.\\\\/foo/i.
"
`;

exports[`CLI accepts exact filenames 3`] = `""`;
